core/ecschema-editing/src/test/Validation/SchemaComparison.test.ts

```diff
@@ -0,0 +1,556 @@
+  };
+
+  const schemaBJson = {
+    $schema: "https://dev.bentley.com/json_schemas/ec/32/ecschema",
+    name: "SchemaB",
+    version: "1.2.3",
+    alias: "b",
+    label: "labelB",
+    description: "descriptionB",
+  };
+
+  beforeEach(async () => {
+    reporter = new TestSchemaCompareReporter();
+    contextA = new SchemaContext();
+    contextB = new SchemaContext();
+  });
+
+  describe("Entity Class comparisons cases", () => {
+    it("should not report baseClass delta when comparing testEntityClass", async () => {
+      const schemaA = await Schema.fromJson({
+        ...schemaAJson,
+        items: {
+          testBaseClass: {
+            schemaItemType: "EntityClass",
+            description: "Test base class",
+          },
+          testEntityClass: {
+            schemaItemType: "EntityClass",
+            baseClass: "SchemaA.testBaseClass",
+          },
+        },
+      }, contextA);
+
+      const schemaB = await Schema.fromJson({
+        ...schemaBJson,
+        items: {
+          testBaseClass: {
+            schemaItemType: "EntityClass",
+            description: "Test base class",
+          },
+          testEntityClass: {
+            schemaItemType: "EntityClass",
+            baseClass: "SchemaB.testBaseClass",
+          },
+        },
+      }, contextB);
+
+      const comparer = new SchemaComparer(reporter);
+      await comparer.compareSchemas(schemaA, schemaB);
+
+      const foundDiag = findDiagnostic(reporter.changes[0].allDiagnostics, "SC-105", "SchemaA.testBaseClass", "SchemaB.testBaseClass");
+      expect(foundDiag).to.equal(false);
+
+    });
+
+    it("should report baseClass delta for testEntityClass", async () => {
+      const schemaA = await Schema.fromJson({
+        ...schemaAJson,
+        items: {
+          testBaseClassA: {
+            schemaItemType: "EntityClass",
+            description: "Test base class A",
+          },
+          testEntityClass: {
+            schemaItemType: "EntityClass",
+            description: "Test entity class",
+            baseClass: "SchemaA.testBaseClassA",
+          },
+        },
+      }, contextA);
+
+      const schemaB = await Schema.fromJson({
+        ...schemaAJson, // Using schemaAJson
+        items: {
+          testBaseClassB: {
+            schemaItemType: "EntityClass",
+            description: "Test base class B",
+          },
+          testEntityClass: {
+            schemaItemType: "EntityClass",
+            description: "Test entity class",
+            baseClass: "SchemaA.testBaseClassB",
+          },
+        },
+      }, contextB);
+
+      const comparer = new SchemaComparer(reporter);
+      await comparer.compareSchemas(schemaA, schemaB);
+
+      const foundDiag = findDiagnostic(reporter.changes[0].allDiagnostics, "SC-105", "SchemaA.testBaseClassA", "SchemaA.testBaseClassB");
+      expect(foundDiag).to.equal(true);
+    });
+
+    it("should not report baseClass delta when comparing testEntityClass", async () => {
```

Comment:
What is the difference between this test and the first test?

Reply:
No difference, removed this test.